### ISSUE TITLE
XML parsing error while retrieving (listing) ResourceRecordSets

### ISSUE THREAD

**Issue by ashutoshpardeshi** [localstack/localstack#3172]
# Type of request: This is a ...
[X ] bug report
[ ] feature request

# Detailed description
I am using localstack to test route53 record creation. One of the records we create is a TXT record containing '&' in the text. While retrieving these records I get the following error:
botocore.parsers.ResponseParserError: Unable to parse response (not well-formed (invalid token): line 19, column 84), invalid XML received. Further retries may succeed:
I have tested this with AWS CLI and it works seemlessly, but with localstack it fails and throw XML error. 
route53_client = boto3.client('route53',
                              endpoint_url='http://localstack:4566',
                              region_name='dummy',
                              aws_access_key_id='dummy',
                              aws_secret_access_key='dummy')

recordsets = route53_client.list_resource_record_sets(HostedZoneId=zone_id,StartRecordType='TXT')

Above line is where the code breaks and throws invalid xml error.

...

## Expected behavior
The list_resource_record_sets api should return all the recordsets in the given hosted zone. 
...

## Actual behavior
Instead , I am getting invalid XML, botocore.parsers.ResponseParserError: Unable to parse response (not well-formed (invalid token): line 19, column 84), invalid XML received. Further retries may succeed:
...

# Steps to reproduce
Create  TXT type recordsets in a given hosted zone using localstack
As a value,  put "ID=xyz&AltId=abc"
Retrieve this record either using boto3 lib api or awslocal awslocal route53 list-resource-record-sets --hosted-zone-id XYZ
It throws the invalid xml error.
` awslocal route53 list-resource-record-sets --hosted-zone-id XYZ

Unable to parse response (not well-formed (invalid token): line 249, column 84), invalid XML received. Further retries may succeed:
b'<ListResourceRecordSetsResponse xmlns="https://route53.amazonaws.com/doc/2012-12-12/">\n   <ResourceRecordSets>\n   \n      <ResourceRecordSet>\n                <Name>abc-t-0-svc.saas.local-test.com.</Name>\n`
....

## Command used to start LocalStack
via docker compose:
docker-compose up -d --build

...

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)
`route53_client = boto3.client('route53',
                              endpoint_url='http://localstack:4566',
                              region_name='dummy',
                              aws_access_key_id='dummy',
                              aws_secret_access_key='dummy')
zone_id = "ZONEID"
recordsets = route53_client.list_resource_record_sets(HostedZoneId=zone_id,StartRecordType='TXT')`

...

┆Issue is synchronized with this [Jira Task](https://localstack.atlassian.net/browse/LOC-81) by [Unito](https://www.unito.io/learn-more)

**Comment by usmangani1** [comment]
Hi @ashutoshpardeshi  Thanks for reporting this issue.With the given information we tried to reproduce the issue with the commands attached below and we are not able to reproduce this issue.

```
$ aws route53 create-hosted-zone --name example.com --caller-reference 2014-04-01-18:47 --endpoint-url=http://localhost:4566
{
    "HostedZone": {
        "Id": "/hostedzone/F7UDY5P18L51D84",
        "Name": "example.com.",
        "Config": {
            "PrivateZone": false
        },
        "ResourceRecordSetCount": 0
    },
    "DelegationSet": {
        "NameServers": [
            "moto.test.com"
        ]
    }
}
$ aws route53 list-resource-record-sets --hosted-zone-id F7UDY5P18L51D84 --endpoint-url=http://localhost:4566
{
    "ResourceRecordSets": []
}
k$ aws route53 change-resource-record-sets --hosted-zone-id F7UDY5P18L51D84 --change-batch file://sample_batch.json --endpoint-url=http://localhost:4566
{
    "ChangeInfo": {
        "Id": "/change/C2682N5HXP0BZ4",
        "Status": "INSYNC",
        "SubmittedAt": "2010-09-10T01:36:41.958Z"
    }
}
$ aws route53 list-resource-record-sets --hosted-zone-id F7UDY5P18L51D84 --endpoint-url=http://localhost:4566
{
    "ResourceRecordSets": [
        {
            "Name": "a.example.com.",
            "Type": "A",
            "TTL": 300,
            "ResourceRecords": [
                {
                    "Value": "4.4.4.4"
                }
            ]
        }
    ]
}

```
Can you please let us know in case if we are doing something wrong in reproducing the issue or please pull the latest version of localstack pro and give it a try ?

**Comment by usmangani1** [comment]
Hi @ashutoshpardeshi  Thanks for reporting this issue.With the given information we tried to reproduce the issue with the commands attached below and we are not able to reproduce this issue.

```
$ aws route53 create-hosted-zone --name example.com --caller-reference 2014-04-01-18:47 --endpoint-url=http://localhost:4566
{
    "HostedZone": {
        "Id": "/hostedzone/F7UDY5P18L51D84",
        "Name": "example.com.",
        "Config": {
            "PrivateZone": false
        },
        "ResourceRecordSetCount": 0
    },
    "DelegationSet": {
        "NameServers": [
            "moto.test.com"
        ]
    }
}
$ aws route53 list-resource-record-sets --hosted-zone-id F7UDY5P18L51D84 --endpoint-url=http://localhost:4566
{
    "ResourceRecordSets": []
}
k$ aws route53 change-resource-record-sets --hosted-zone-id F7UDY5P18L51D84 --change-batch file://sample_batch.json --endpoint-url=http://localhost:4566
{
    "ChangeInfo": {
        "Id": "/change/C2682N5HXP0BZ4",
        "Status": "INSYNC",
        "SubmittedAt": "2010-09-10T01:36:41.958Z"
    }
}
$ aws route53 list-resource-record-sets --hosted-zone-id F7UDY5P18L51D84 --endpoint-url=http://localhost:4566
{
    "ResourceRecordSets": [
        {
            "Name": "a.example.com.",
            "Type": "TXT",
            "TTL": 300,
            "ResourceRecords": [
                {
                    "Value": "4.4.4.4"
                }
            ]
        }
    ]
}

```
Can you please let us know in case if we are doing something wrong in reproducing the issue or please pull the latest version of localstack pro and give it a try ?

**Comment by ashutoshpardeshi** [comment]
Hello,
So if you add an ampersand "&" in the TXT record value that will break. Can
you try that?

Thank you,

**Comment by usmangani1** [comment]
@ashutoshpardeshi Can you please help us with the sample json file specifying the resources which are being changed here.
Thanks

**Comment by bblommers** [comment]
@whummer I haven't verified this against Localstack, but I believe this is now fixed upstream as of moto >= 2.2.18.dev21

**Comment by whummer** [comment]
Great, thanks for the pointer @bblommers - did a quick manual check, and this seems to be fixed in the meantime.. 👍  Thanks!
